### DISCUSSION TITLE
When generate Registration Options, can we enable only the "my devices"

### DISCUSSION THREAD

**Discussion by alan345** [MasterKale/SimpleWebAuthn#407] [Troubleshooting]
Hello, 

Can we hide the all other options in this screen and keep only "This device"?
<img width="451" alt="image" src="https://github.com/MasterKale/SimpleWebAuthn/assets/15246526/af6e1cf1-b721-4d12-9e77-d38845aa6c5d">

When signup the passkey in stripe.com and cvs.com and, it did give me the options to setup the passkey only with my current device.

Thanks for your help!

**Comment by MasterKale** [comment]
You sure can, you just need to specify the `authenticatorSelection` option when calling `generateRegistrationOptions()` to specify you're only interested in "platform" authenticators:

```ts
const options = generateRegistrationOptions({
  // ...
  authenticatorSelection: {
    authenticatorAttachment: 'platform',
  },
});
```

![Screenshot 2023-07-05 at 1 31 24 PM](https://github.com/MasterKale/SimpleWebAuthn/assets/5166470/9acc4c8e-6873-4cd6-90f4-8b23eba5d22f)

The other option, `"cross-platform"`, would narrow down the list as well, but only show the security key and "hybrid"/"cross-device" options:

![Screenshot 2023-07-05 at 1 31 36 PM](https://github.com/MasterKale/SimpleWebAuthn/assets/5166470/3ddb2419-33b0-4519-8b6b-b4982f68a3cb)

Here's a bit more info about the two possible values if you want to read up about them: https://www.w3.org/TR/webauthn-2/#enum-attachment

**Reply by alan345** [reply]
Awesome! thanks!

**Reply by alan345** [reply]
You should add a quick comment here: https://simplewebauthn.dev/docs/packages/server#1-generate-registration-options

**Reply by alan345** [reply]
I created a PR https://github.com/MasterKale/SimpleWebAuthn-homepage/pull/33

**Comment by alan345** [comment]
It works well on the Registration (`generateRegistrationOptions `) but now I am having the issue on the Authentication (`generateAuthenticationOptions`)

Do we have `authenticatorAttachment: 'platform'` for the Authentication?


I registered with my phone and when I login with my desktop, I get this
<img width="449" alt="image" src="https://github.com/MasterKale/SimpleWebAuthn/assets/15246526/7fa115f4-322f-4097-89ff-14af6ffd70d3">
(There are no passkey setup on the desktop)



Thanks

**Reply by MasterKale** [reply]
Just to double-check, are you passing `transports` in with the entries you include in `allowCredentials`? Those can go a long way in helping the **browser** understand how it might communicate with the authenticator that'd recognize the credential ID, and browsers will remove from their lists any options that don't make sense. That's the best an RP can do during authentication unfortunately...there's nothing like `authenticatorAttachment` for authentications.
